subsys/bluetooth/audio/tbs.c

```diff
@@ -310,15 +310,18 @@ static uint8_t next_free_call_index(void)
 		static uint8_t next_call_index = 1;
 		const struct bt_tbs_call *call = lookup_call(next_call_index);
 
-		if (call == NULL) {
-			return next_call_index++;
-		}
-
-		next_call_index++;
 		if (next_call_index == UINT8_MAX) {
 			/* call_index = 0 reserved for outgoing calls */
 			next_call_index = 1;
 		}
+
+		/* For each new call, the call index should be incremented */
+		next_call_index++;
+
+		if (call == NULL) {
+			/* call == NULL is true for (next_call_index - 1)*/
+			return (next_call_index - 1);
+		}
```

Comment:
Either I'm missing something, or this won't return 255

Reply:
Yes, you are right.

Reply:
Should be fixed now. @MariuszSkamra